pkg/api/api.go

```diff
@@ -68,6 +76,86 @@ func (a *ScratchDataAPIStruct) AuthGetDatabaseID(ctx context.Context) int64 {
 	return ctx.Value("databaseId").(int64)
 }
 
+func (a *ScratchDataAPIStruct) CreateQuery(w http.ResponseWriter, r *http.Request) {
+	apiKey := r.URL.Query().Get("api_key")
```

Comment:
You do not need to do any API key validation here because it is all done in`AuthMiddleware`